mypy/plugins/dataclasses.py

```diff
@@ -74,6 +76,7 @@
     frozen_default=False,
     field_specifiers=("dataclasses.Field", "dataclasses.field"),
 )
+_INTERNAL_REPLACE_SYM_NAME = "__mypy_replace"
```

Comment:
I would propose to use a syntactically invalid name for internal things, like `mypy-replace`, we do this in several other places, see e.g. `get_unique_redefinition_name()`.

Reply:
Nice. Good idea using an invalid name.

Reply:
I had a similar need in my pynamodb plugin.

Reply:
See `deserialize_type` call:
https://github.com/pynamodb/pynamodb-mypy/blob/a9684e20b413c39b447e54238003c2d99a5c0c73/pynamodb_mypy/plugin.py#L50-L60

... even though... I can't quite remember, maybe that's a good thing? i.e.maybe due to caching semanal is skipped and `metadata` **is** what I'm supposed to consult?

But, as you can see, in the checker phase I don't have a public API to really "deserialize" what's in the metadata back into a type.